components/collection/drop/PaidGenerative.vue

```diff
@@ -95,28 +100,13 @@ const mintNft = async () => {
     loading.value = true
     mintingSession.value.txHash = undefined
 
-    const { apiInstance } = useApi()
-    const api = await apiInstance.value
-
-    initTransactionLoader()
-
-    const cb = api.tx.utility.batchAll
-    const args = allocatedNFTs.value.map((allocatedNFT) =>
-      api.tx.nfts.mint(
-        drop.value?.collection,
-        allocatedNFT.id,
-        accountId.value,
-        {
-          ownedItem: null,
-          mintPrice: drop.value?.price,
-        },
-      ),
-    )
-
-    howAboutToExecute(accountId.value, cb, [args], {
-      onResult: ({ txHash }) => {
-        mintingSession.value.txHash = txHash
-      },
+    transaction({
+      interaction: NFTs.MINT_DROP,
+      collectionId: drop.value?.collection,
+      nfts: allocatedNFTs.value.map((allocatedNft) => ({
+        id: allocatedNft.id,
+      })),
```

Comment:
just past `allocatedNFTs.value` here? can save redundant `map` calls

same with `HolderOfGenerative.vue`

```suggestion
      nfts: allocatedNFTs.value,
```

Reply:
Okay, it's doable.